ui/src/app/utils/cluster-initializer.tsx

```diff
@@ -0,0 +1,301 @@
+import {notebooksClusterApi} from 'app/services/notebooks-swagger-fetch-clients';
+import {clusterApi} from 'app/services/swagger-fetch-clients';
+import {isAbortError, reportError} from 'app/utils/errors';
+import {Cluster, ClusterStatus} from 'generated/fetch';
+
+// We're only willing to wait 20 minutes total for a cluster to initialize. After that we return
+// a rejected promise no matter what.
+const DEFAULT_OVERALL_TIMEOUT = 1000 * 60 * 20;
```

Comment:
Can you use this `Duration` class? https://nodes.duniter.io/typescript/duniter/typedoc/interfaces/moment.duration.html